### PR TITLE
Update the handler to use zap logger

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Previously the handler was using hclog, which logs the webhook logs to a different output from the controller logs. In order for them to be written to where the controller logs are, we update the handler to now use zaplogger over hclog.

How I've tested this PR: Run tests

How I expect reviewers to test this PR: Code review

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: subcommand/connect-init/command.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ const (
 	// The number of times to attempt ACL Login.
 	numLoginRetries = 3
 	// The number of times to attempt to read this service (60s).
-	defaultServicePollingRetries = 60
+	defaultServicePollingRetries = 120
```
Comment (ishustava) [inline subcommand/connect-init/command.go:27]: this change seems unrelated?

File: subcommand/inject-connect/command.go [hunk 2]
```diff
@@ -351,8 +346,14 @@ func (c *Command) Run(args []string) int {
 	allowK8sNamespaces := flags.ToSet(c.flagAllowK8sNamespacesList)
 	denyK8sNamespaces := flags.ToSet(c.flagDenyK8sNamespacesList)
 
-	// Start the endpoints controller
-	zapLogger := zap.New(zap.UseDevMode(true), zap.Level(zapcore.InfoLevel))
+	var zapLevel zapcore.Level
+	if err := zapLevel.UnmarshalText([]byte(c.flagLogLevel)); err != nil {
+		c.UI.Error(fmt.Sprintf("Error parsing -log-level %q: %s", c.flagLogLevel, err.Error()))
+		return 1
+	}
```
Comment (ishustava) [inline subcommand/inject-connect/command.go:353]: does zaplogger support the same levels as hclog?

Comment (thisisnotashwin) [reply subcommand/inject-connect/command.go:353]: good catch..trace is not supported!! updating the PR!

File: subcommand/connect-init/command.go (hunk 1 above)
Comment (thisisnotashwin) [reply subcommand/connect-init/command.go:27]: yeah..i just crammed it into this PR but I can remove it from this one and add another PR for the same

File: subcommand/inject-connect/command.go (hunk 2 above)
Comment (thisisnotashwin) [reply subcommand/inject-connect/command.go:353]: Updated the flag description to only include the levels supported by zaplogger!

File: connect-inject/handler.go [hunk 3]
```diff
@@ -198,15 +198,15 @@ func (h *Handler) Handle(_ context.Context, req admission.Request) admission.Res
 	// the Envoy configuration.
 	initContainer, err := h.containerInit(pod, req.Namespace)
 	if err != nil {
```
Comment (ishustava) [inline connect-inject/handler.go:200]: since we're changing it, I wonder if we should add an `Info` level log line saying the name/namespace of the pod it's mutating, maybe on line 176?

**Review by ishustava — APPROVED** [review]

File: connect-inject/handler.go (hunk 3 above)
Comment (thisisnotashwin) [reply connect-inject/handler.go:202]: Done!! 😺

**Review by kschoche — APPROVED** [review]
Excellent!
